experiment:
    save_to: /home/tartandriver/workspace/experiments/yamaha_irl_dino/
    name: irl_radiov3_siglip2_voxel_recolor
    device: cuda
    epochs: 100
    steps_per_epoch: 100
    save_every: 1

footprint:
    params:
        length: 4.0
        width: 2.0
          
        #displacement to get to the center of the footprint from the center of the rear axle
        length_offset: 1.2
        width_offset: 0.0
        
        nl: 25
        nw: 15

algo:
    type: MPPIIRLSpeedmaps
    params:
        batch_size: 1
        reg_coeff: 0.0000001
        # reg_coeff: 0.0 #turn off speedmaps for now
        speed_coeff: 1.0
        grad_clip: 10.0

#TODO nest configs
dataset:
    irl:
        min_speed: 1.0
        sample_every: 20

    common:
        # root_dir: /media/airlab/ssd1/2025_irl_kitti/train
        root_dir: /media/charlie/extreme_ssd/2025_irl_kitti/train
        device: cuda
        use_multiprocessing: true

    datatypes:
        bev_data:
            dtype: BEVGrid
            dir: bev_map_reduce
            
            args: {}

        coord_voxel_data:
            dtype: VoxelGrid
            dir: coord_voxel_map

            args:
                subtract_curr_height: false

        odometry:
            dtype: OdomRBState
            dir: odometry
            temporal_sample: [0, 76]

            args: {}

        steer_angle:
            dtype: Float32
            dir: steer_angle
            temporal_sample: [0, 76]

            args: {}

        feature_image:
            dtype: FeatureImage
            dir: radio3_siglip2_feature_image_full

            args: {}

        feature_image_intrinsics:
            dtype: Intrinsics
            dir: feature_image_intrinsics

            args: {}

        image:
            dtype: Image
            dir: image

            args: {}

        image_intrinsics:
            dtype: Intrinsics
            dir: image_intrinsics

            args: {}

    transforms:
        tf_odom_to_cam:
            src_data: image
            src_frame: sensor_init
            dst_frame: multisense/left_camera_optical_frame

network:
    type: VoxelRecolorBEVToCostSpeed

    args:
        terrain_layer: terrain
        bev_layers: ['num_voxels', 'slope', 'diff', 'min_elevation_filtered_inflated_mask']

        recolor_params:
            coord_vg_key: 'coord_voxel_data'
            img_keys: ['radio3_siglip2_feature_image_full']

            feat_net_params:
                hidden_channels: [128, 128]
                hidden_activation: relu
                out_channels: 64

        resnet_params:
            hidden_channels: [128,]
            hidden_activation: relu

        cost_params:
            type: Categorical
            output_params:
                bounds: [0., 5.]
                nbins: 20

            net_params:
                hidden_channels: []
                hidden_activation: relu

        speed_params:
            type: Categorical
            output_params:
                bounds: [0., 15.]
                nbins: 15
                
            net_params:
                hidden_channels: []
                hidden_activation: relu

netopt:
    type: Adam
    params: {}

solver:
    type: mpc
    mpc_fp: /home/tartandriver/tartandriver_ws/src/planning/maxent_irl_maps/config/mpc/throttle_mppi.yaml